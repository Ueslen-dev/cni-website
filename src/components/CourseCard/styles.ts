import styled from 'styled-components';

export const Wrapper = styled.section`
  -webkit-box-shadow: -2px 2px 16px 1px #b0b0b0;
  box-shadow: -2px 2px 16px 1px #b0b0b0;
  background-color: var(--white-color);
  padding: 20px;
  margin-bottom: 20px;
  border-radius: 5px;
  cursor: pointer;
`;

export const Image = styled.figure`
  img {
    border-radius: 5px;
  }
`;

export const Infos = styled.aside`
  margin-top: 20px;
  h1 {
    font-size: 2.2rem;
    margin-bottom: 20px;
  }
`;

export const Teacher = styled.div`
  display: flex;
  align-items: center;
  gap: 1rem;
  color: var(--grey-color);
  margin-bottom: 20px;
  .anticon-user {
    font-size: 20px;
  }
  span {
    font-size: 1.6rem;
    font-weight: 600;
  }
`;

export const Price = styled.div`
  display: flex;
  align-items: center;
  gap: 2rem;
  span {
    font-weight: bold;
  }
  span:first-child {
    color: var(--blue-color);
    font-size: 3rem;
  }
  span:last-child {
    color: var(--grey-color);
    font-size: 2rem;
    text-decoration: line-through;
  }
`;
