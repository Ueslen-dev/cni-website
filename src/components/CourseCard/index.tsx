import { UserOutlined } from '@ant-design/icons';

import * as S from './styles';

type Props = {
  image: string;
  alt: string;
  name: string;
  teacher: string;
  price: string;
  offer?: string;
};

const CourseCard = ({ image, alt, name, teacher, price, offer }: Props) => (
  <S.Wrapper>
    <S.Image>
      <img src={image} alt={alt} loading="lazy" />
    </S.Image>
    <S.Infos>
      <h1>{name}</h1>
      <S.Teacher>
        <UserOutlined />
        <span>{teacher}</span>
      </S.Teacher>
      <S.Price>
        <span>R${offer ? offer : price}</span>
        <span>R${offer && price}</span>
      </S.Price>
    </S.Infos>
  </S.Wrapper>
);
export default CourseCard;
