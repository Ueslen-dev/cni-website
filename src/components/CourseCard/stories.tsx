import { Story, Meta } from '@storybook/react/types-6-0';

import CourseCard from '.';

export default {
  title: 'CourseCard',
  component: CourseCard,
  args: {
    image: 'http://www.esmaltale.com.br/img/mEp.png',
    alt: 'Mulher pintando a unha do pé e a unha da mão',
    name: 'Curso de manicure',
    teacher: 'Aline Sila Campos',
    price: '300,00',
    offer: '230,00'
  }
} as Meta;

export const Basic: Story = (args) => {
  const { image, alt, name, teacher, price, offer } = args;
  return (
    <CourseCard
      image={image}
      alt={alt}
      name={name}
      teacher={teacher}
      price={price}
      offer={offer}
    />
  );
};
