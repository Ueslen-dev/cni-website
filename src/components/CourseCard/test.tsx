import { screen, render } from '@testing-library/react';

import CourseCard from '.';

describe('<CourseCard/>', () => {
  it('should render the heading', () => {
    const { container } = render(<CourseCard />);
    expect(
      screen.getByRole('heading', { name: /CourseCard/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

});
