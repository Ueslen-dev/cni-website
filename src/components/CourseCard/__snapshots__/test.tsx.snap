// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseCard/> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    CourseCard
  </h1>
</main>
`;
